File: pm_core/src/infrastructure/store/mapper/issue_type_scheme_mapper.go
package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type IssueTypeSchemeMapper struct{}

func NewIssueTypeSchemeMapper() *IssueTypeSchemeMapper {
	return &IssueTypeSchemeMapper{}
}

func (m *IssueTypeSchemeMapper) ToEntity(mdl *model.IssueTypeSchemeModel) *entity.IssueTypeSchemeEntity {
	if mdl == nil {
		return nil
	}
	return &entity.IssueTypeSchemeEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:           mdl.TenantID,
		Name:               mdl.Name,
		Description:        mdl.Description,
		DefaultIssueTypeID: mdl.DefaultIssueTypeID,
	}
}

func (m *IssueTypeSchemeMapper) ToModel(e *entity.IssueTypeSchemeEntity) *model.IssueTypeSchemeModel {
	if e == nil {
		return nil
	}
	return &model.IssueTypeSchemeModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			CreatedBy: e.CreatedBy,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:           e.TenantID,
		Name:               e.Name,
		Description:        e.Description,
		DefaultIssueTypeID: e.DefaultIssueTypeID,
	}
}

func (m *IssueTypeSchemeMapper) ToEntities(models []*model.IssueTypeSchemeModel) []*entity.IssueTypeSchemeEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.IssueTypeSchemeEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}
	return entities
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/IssueTypeSchemeMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.IssueTypeSchemeEntity;
import serp.project.pmcore.infrastructure.store.model.IssueTypeSchemeModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class IssueTypeSchemeMapper extends BaseMapper {

    public IssueTypeSchemeEntity toEntity(IssueTypeSchemeModel model) {
        if (model == null) { return null; }
        return IssueTypeSchemeEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .name(model.getName())
                .description(model.getDescription())
                .defaultIssueTypeId(model.getDefaultIssueTypeId())
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public IssueTypeSchemeModel toModel(IssueTypeSchemeEntity entity) {
        if (entity == null) { return null; }
        return IssueTypeSchemeModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .name(entity.getName())
                .description(entity.getDescription())
                .defaultIssueTypeId(entity.getDefaultIssueTypeId())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<IssueTypeSchemeEntity> toEntities(List<IssueTypeSchemeModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }
}
